### ISSUE TITLE
Consolidate lib directories

### ISSUE THREAD

**Issue by agarwal** [ocaml/v3.ocaml.org-rescript#141]
We have several directories with pure ReScript code (by "pure" I mean non-pages files): `bindings`, `common`, `components`, and `layouts`. It feels like too much categorization. I think we could simplify this into a single directory.

**Comment by ghost** [comment]
Mind the tailwind config. There is a purge section which is sensitive to these locations.

**Comment by ghost** [comment]
Some getStaticProps implementations refer to directory names

**Comment by ghost** [comment]
Let's finish this sooner, since more people are getting familiar with the directory layout currently.

**Comment by agarwal** [comment]
Yes, I want to do this next, immediately after merging my current PR.

**Comment by ghost** [comment]
The directory layout below shows one example of a directory layout for a large, real world nextjs project.

![image](https://user-images.githubusercontent.com/1385802/116761966-0157c180-a9e7-11eb-9cd0-49a2d6d2cbee.png)

**Comment by ghost** [comment]
@agarwal It's useful having `components` in a subdirectory because we aren't using StoryBook currently, so browsing the contents of a `components` directory is one quick way to find the available reusable components when implementing a new page.

**Comment by rdavison** [comment]
Merged: https://github.com/ocaml/v3.ocaml.org/pull/346

> @agarwal It's useful having components in a subdirectory because we aren't using StoryBook currently, so browsing the contents of a components directory is one quick way to find the available reusable components when implementing a new page.

One way to answer that question would be to generate documentation for the library and look for all values that produce `React.element`s.

**Comment by agarwal** [comment]
> It's useful having `components` in a subdirectory because we aren't using StoryBook currently

We've merged all into a single directory for now. Let's see how we feel after a while. I do think if we create separate directories, it indicates you're seeking a namespace. Thus, my inclination is to then make a separate library. Also, we do certainly need to write documentation that gives an overview of the modules, and every module should have documentation at the top if its `.resi` file. Both are currently missing, so we have more work to do.

I QA'd also. Everything seems to work. Closing this issue.
